drools-impact-analysis-graph/drools-impact-analysis-graph-common/src/main/java/org/drools/impact/analysis/graph/Node.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.drools.impact.analysis.graph;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class Node {
+
+    private String packageName;
+
+    private String ruleName;
+
+    private Set<Link> incomingLinks;
+    private Set<Link> outgoingLinks;
```

Comment:
The graph objects are only Node and Link. At the moment, I use only `outgoingLinks` but there is a requirement of backward relationship view so I also maintains incomingLinks.